Sources/WalletConnectNotify/Client/Wallet/NotifyWatchAgreementService.swift

```diff
@@ -0,0 +1,54 @@
+import Foundation
+
+final class NotifyWatchAgreementService {
```

Comment:
I am not sure I like the name of the class 😅
what about NotifyWatcherAgreementKeysProvider NotifySubscriptionsWatcherKeysProvider?